### PR TITLE
Create initial project structure

### PR DESCRIPTION
**Pull request by BraydonKains**
This commit adds the currently agreed upon project structure with
independent packages for data, auth, and graphql. It also adds editor
config support and a Makefile for building the GraphQL server.

### CODE REVIEW

File: graphql/server.go [hunk 1]
```diff
@@ -45,7 +29,7 @@ func (s *server) registerEchoMutation(schema *schemabuilder.Schema) {
 // registerGame registers the game type.
 func (s *server) registerGame(schema *schemabuilder.Schema) {
```
Comment (zdylag) [inline graphql/server.go:30]: Long term when we actually have stuff here I bet this also breaks into a bunch of files.

File: graphql/server_test.go [hunk 2]
```diff
@@ -115,9 +118,111 @@ func TestGamesFiltering(t *testing.T) {
 	}
 }
 
+func TestUsersListAll(t *testing.T) {
```
Comment (zdylag) [inline graphql/server_test.go:121]: Adding tests? 12/10!

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.PHONY: graphql
+gql:
+	go build ./graphql/main.go
```
Comment (zdylag) [inline Makefile:3]: Do we want this to run? Unsure 🤷‍♀️ .

**Review by zdylag — COMMENTED** [review]
Some final commit nits then I think it seems great!

In the commit Makefile is missing its last letter.
Can we prefix the commit title with `go: ` or `*: `?

File: Makefile (hunk 3 above)
Comment (BraydonKains) [reply Makefile:3]: I just started it this way cause my personal flow has always been "build then run" and very rarely using `go run`, but I think it's perfectly fine to have a run option as well.

File: graphql/server.go (hunk 1 above)
Comment (BraydonKains) [reply graphql/server.go:30]: I agree. Decided not to go crazy yet, but I hope it gets split into more files as it scales.

File: graphql/main.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package graphql
```
Comment (zdylag) [inline graphql/main.go:1]: I finally checked this out to notice, but you cannot run non-main packages. Often the way you get around this is introducing an app folder that has small subdirectories each with a tiny "main.go".

File: app/graphql_server/main.go [hunk 5]
```diff
@@ -27,13 +28,13 @@ func main() {
 		{Runner: users[0], Game: games[0], Time: 17 * time.Millisecond},
 		{Runner: users[1], Game: games[0], Time: 3 * time.Hour},
 	}
-	server := &(server{
-		games: games,
-		users: users,
-		runs:  runs,
+	server := &(graphql_server.Server{
```
Comment (zdylag) [inline app/graphql_server/main.go:31]: Jsyk: the `(` aren't needed here.

**Review by zdylag — APPROVED** [review]
I confirmed the tests pass and I can run the server 👍

File: app/graphql_server/main.go (hunk 5 above)
Comment (BraydonKains) [reply app/graphql_server/main.go:31]: Neat, I'm not sure why I did this by reflex. The next person to touch this file can fix (or if I need to make fixes I'll remove them)

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*.go,Makefile]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
```
Comment (hyperflat) [inline .editorconfig:7]: tabs are required in Makefiles

Comment (zdylag) [reply .editorconfig:7]: I wonder if this editorconfig isn't taking effect in the Makefile? It doesn't seem to do anything? (I think I installed the correct extension).

**Review by JoshSanch — APPROVED** [review]
As far as `server.go`, it seems like there might be a way to automate endpoint/test generation based on a configurable set of fields for our data models. That would take some work though, and this is currently a solid start for an MVP.

File: .editorconfig (hunk 6 above)
Comment (hyperflat) [reply .editorconfig:7]: yes, it's not expanded. fixed in #26

**Pull request merged**